src/NLog/Config/LoggingConfigurationParser.cs

```diff
@@ -184,78 +184,61 @@ private void SetNLogElementSettings(Dictionary<string, string> dict, out bool? p
-            if (dict.TryGetValue("INTERNALLOGLEVEL", out var val))
-            {
-                // expanding variables not possible here, they are created later
-                InternalLogger.LogLevel = ParseLogLevelSafe("InternalLogLevel", val, InternalLogger.LogLevel);
-                internalLoggerEnabled = InternalLogger.LogLevel != LogLevel.Off;
-            }
-            else
-            {
-                internalLoggerEnabled = false;
-            }
 
-            return internalLoggerEnabled;
-        }
-
-        /// <summary>
-        /// Set <see cref="LogFactory.ThrowConfigExceptions"/>
-        /// </summary>
-        /// <param name="dict"></param>
-        private void SetThrowConfigExceptions(IDictionary<string, string> dict)
-        {
-            if (dict.TryGetValue("THROWCONFIGEXCEPTIONS", out var val))
+            if (!internalLoggerEnabled && !InternalLogger.HasActiveLoggers())
             {
-                LogFactory.ThrowConfigExceptions = StringHelpers.IsNullOrWhiteSpace(val)
-                    ? (bool?)null
-                    : ParseBooleanValue("ThrowConfigExceptions", val, false);
+                InternalLogger.LogLevel = LogLevel.Off; // Reduce overhead of the InternalLogger when not configured
             }
-        }
 
-        /// <summary>
-        /// Set <see cref="LogFactory.ThrowExceptions"/>
-        /// </summary>
-        /// <param name="dict"></param>
-        private void SetThrowExceptions(IDictionary<string, string> dict)
-        {
-            if (dict.TryGetValue("THROWEXCEPTIONS", out var val))
-            {
-                LogFactory.ThrowExceptions = ParseBooleanValue("ThrowExceptions", val, LogFactory.ThrowExceptions);
-            }
+            _configurationItemFactory = ConfigurationItemFactory.Default;
+            _configurationItemFactory.ParseMessageTemplates = parseMessageTemplates;
         }
 
         /// <summary>
-        /// build dictionary, use last value of duplicates
+        /// Builds list with unique keys, using last value of duplicates. High priority keys placed first.
         /// </summary>
         /// <param name="nlogConfig"></param>
         /// <returns></returns>
-        private static Dictionary<string, string> CreateNLogConfigDictionary(ILoggingConfigurationElement nlogConfig)
+        private static List<KeyValuePair<string,string>> CreateUniqueSortedListFromConfig(ILoggingConfigurationElement nlogConfig)
         {
             var dict = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
             foreach (var configItem in nlogConfig.Values)
             {
-                if (configItem.Key != null)
+                if (!string.IsNullOrEmpty(configItem.Key))
                 {
                     dict[configItem.Key.Trim()] = configItem.Value;
                 }
             }
 
-            return dict;
+            var sortedDict = new List<KeyValuePair<string, string>>(dict.Count);
```

Comment:
List instead of dict?

Reply:
The order is predictable for List, not always for Dictionary.

Reply:
I mean the name of the variable ;)

Reply:
Fixed